cpp/src/io/parquet/parquet_gpu.hpp

```diff
@@ -284,6 +291,7 @@ struct file_intermediate_data {
   hostdevice_vector<gpu::PageInfo> pages_info{};
   hostdevice_vector<gpu::PageNestingInfo> page_nesting_info{};
   hostdevice_vector<gpu::PageNestingDecodeInfo> page_nesting_decode_info{};
+  rmm::device_buffer level_decode_data;
```

Comment:
should we use `device_uvector<uint32_t>` here?

Reply:
I just changed the type here to be 16 bit (`level_t`) and after some discussion with @etseidl  offline, we can imagine a use case in the future where this may be variable in size based on the schema, so a device_buffer gives us some wiggle room/